pkg/controller/job/indexed_job_utils_test.go

```diff
@@ -77,6 +80,7 @@ func TestCalculateSucceededIndexes(t *testing.T) {
 			},
 			completions:   4,
 			wantIntervals: []interval{{0, 2}},
+			trackingWithFinalizers: false,
```

Comment:
@alculquicondor As I am starting to remove test code that depends on non finalizer code, I discovered this one case.  If I flip this "trackingWithFinalizers" to true, the test case fails.  I'm unsure if this is a bug and we can fix it by changing the tests or this shows a case that we need to look into.